index.bs

```diff
@@ -217,27 +217,27 @@ interface ImageCapture {
 # PhotoCapabilities # {#photocapabilities-section}
 
 <pre class="idl">
-  [Exposed=Window] interface PhotoCapabilities {
-    readonly attribute RedEyeReduction            redEyeReduction;
-    readonly attribute MediaSettingsRange         imageHeight;
-    readonly attribute MediaSettingsRange         imageWidth;
-    readonly attribute FrozenArray&lt;FillLightMode> fillLightMode;
+  dictionary PhotoCapabilities {
+    RedEyeReduction         redEyeReduction;
+    MediaSettingsRange      imageHeight;
+    MediaSettingsRange      imageWidth;
+    sequence&lt;FillLightMode> fillLightMode;
```

Comment:
Might this deter partial implementations, e.g. shipping fillLightMode detection before redEyeReduction detection?

The reason this came up for me in this review and not mediacapture-main is that here it was an interface before, and moving from interface semantics to dictionary semantics begs the question of whether absence is permitted.

Reply:
It seems like this pattern should/may be used as well for [`ConstrainPoint2DParameters`](https://w3c.github.io/mediacapture-image/#dictdef-constrainpoint2dparameters), [`Point2D`](https://w3c.github.io/mediacapture-image/#dictdef-point2d), and [`PhotoSettings`](https://w3c.github.io/mediacapture-image/#dictdef-photosettings) dictionaries then. Shall we create another PR for this to discuss this as this is the first time I see this pattern?
cc @domenic for thoughts as well

Reply:
No, because constraints are inputs, and that would require JS always passing in both/all members.

In the interest of incrementing, I'd like to take back the `required` recommendation above, as I have some questions around it and it has low value. We can always add it later, as it's largely editorial and for readability.

Reply:
Web IDL states:

> Note that specifying dictionary members as required only has an observable effect when converting other representations of dictionaries (like an ECMAScript value supplied as an argument to an operation) to an IDL dictionary. Specification authors should leave the members optional in all other cases, including when a dictionary type is used solely as the return type of operations.